cmd/oceancenter/main.go

```diff
@@ -0,0 +1,136 @@
+/*
+AUTHORS
+  Alan Noble <alan@ausocean.org>
+
+LICENSE
+  Copyright (C) 2024 the Australian Ocean Lab (AusOcean)
+
+  This file is part of Ocean Center. Ocean Center is free software: you can
+  redistribute it and/or modify it under the terms of the GNU
+  General Public License as published by the Free Software
+  Foundation, either version 3 of the License, or (at your option)
+  any later version.
+
+  Ocean Center is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with Ocean Center in gpl.txt.  If not, see
+  <http://www.gnu.org/licenses/>.
+*/
+
+// Ocean Center is a cloud service for remote device management, including:
+//
+// - device software installation
+// - device software upgrades
+// - device enabling and disabling
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"strconv"
+	"sync"
+
+	"github.com/ausocean/cloud/model"
+	"github.com/ausocean/openfish/datastore"
+)
+
+const (
+	projectID = "oceancenter"
+	version   = "v0.1.0"
+)
+
+// service defines the properties of our web service.
+type service struct {
+	setupMutex    sync.Mutex
+	settingsStore datastore.Store
+	debug         bool
+	standalone    bool
+}
+
+// app is an instance of our service.
+var app *service = &service{}
```

Comment:
Does this even need to be global? No where else uses it besides main; we could just define it in there.

Reply:
It needs to be global, since in GAE `main` is only called once, and is not called subsequently on warm startups.

Reply:
But ```main``` is not returned from right ? ```http.ListenAndServe``` is blocking so the service value will still exist. IIUC for each new instance ```main``` is called.